server/src/contact/contact.service.ts

```diff
@@ -54,14 +54,34 @@ export class ContactService {
 		return null;
 	}
 
+	/**
+	 * Attempt to create a new contact data entry, if it already exists just update it
+	 * @param guest The primaryData of the guest to add contact info for
+	 * @param contact The new or updated contact info
+	 * @return The resulting contact data
+	 */
+	async createOrUpdate(
+		guest: primaryData,
+		contact: contactData,
+	): Promise<contactData> {
+		try {
+			return await this.create(guest, contact);
+		} catch (err) {
```

Comment:
is there a specific error that’s thrown back by typeorm here if specifically the resource already exists? like a 403 or 409 or something?

Reply:
It returns QueryFailedError. I can check for that specifically if you want

Reply:
if they don’t further specify that it’s due to the resource existing already, it’s not much help to check.

Reply:
fair